src/managers/kyuubi.py

```diff
@@ -7,39 +7,75 @@
 from config.hive import HiveConfig
 from config.kyuubi import KyuubiConfig
 from config.spark import SparkConfig
-from core.domain import (
-    DatabaseConnectionInfo,
-    S3ConnectionInfo,
-    SparkServiceAccountInfo,
-    ZookeeperInfo,
-)
+from core.context import Context
 from core.workload import KyuubiWorkloadBase
 from utils.logging import WithLogging
 
 
 class KyuubiManager(WithLogging):
     """Kyuubi manager class."""
 
-    def __init__(self, workload: KyuubiWorkloadBase):
+    def __init__(self, workload: KyuubiWorkloadBase, context: Context):
         self.workload = workload
+        self.context = context
+
+    def _compare_and_update_file(self, content: str, file_path: str) -> bool:
+        """Update the file at given file_path with given content.
+
+        Before doing the update, compare the existing content of the file and update
+        it only if has changed.
+
+        Return True if the file was re-written, else False.
+        """
+        try:
+            existing_content = self.workload.read(file_path)
+        except FileNotFoundError:
+            existing_content = ""
+        self.logger.debug(f"{file_path=}")
+        self.logger.debug(f"{existing_content=}")
+        self.logger.debug(f"{content=}")
+        if existing_content != content:
+            self.workload.write(content, file_path)
+            return True
+
+        return False
 
     def update(
         self,
-        s3_info: S3ConnectionInfo | None,
-        metastore_db_info: DatabaseConnectionInfo | None,
-        auth_db_info: DatabaseConnectionInfo | None,
-        service_account_info: SparkServiceAccountInfo | None,
-        zookeeper_info: ZookeeperInfo | None,
+        set_s3_none: bool = False,
+        set_metastore_db_none: bool = False,
+        set_auth_db_none: bool = False,
+        set_service_account_none: bool = False,
+        set_zookeeper_none: bool = False,
```

Comment:
**praise** Yes, I really like the new approach here! It does simplify and make the code much much much leaner! I really didn't like that changing the signature here would have pretty much effect on the entire code base! WELL DONE!